src/test/ui/uninit-empty-types.rs

```diff
@@ -3,15 +3,18 @@
 
 // pretty-expanded FIXME #23616
 
-use std::mem;
+use std::mem::MaybeUninit;
 
 #[derive(Clone)]
 struct Foo;
 
 #[allow(deprecated)]
 pub fn main() {
     unsafe {
-        let _x: Foo = mem::uninitialized();
-        let _x: [Foo; 2] = mem::uninitialized();
+        // `Foo` and `[Foo; 2]` are both zero sized so this is safe.
```

Comment:
That's not a sufficient argument: `!` is ZST and certainly not safe to create this way.